pkg/kubelet/pleg/generic_test.go

```diff
@@ -736,3 +736,155 @@ kubelet_running_pods 2
 		})
 	}
 }
+
+func TestWatchConditions(t *testing.T) {
```

Comment:
It seems like we could add more tests, e.g.,
 - Multiple containers in the pod
 - Condition not cleared when evaluating an older version

Reply:
Done.